### ISSUE TITLE
Implement Training Loop

### ISSUE THREAD

**Issue by Akettle44** [Akettle44/SRGAN-PyTorch#4]
Goal: Create training class that can be used to train the model.
Features: 
- Training / Validation loop with specified metric
- Ability to load hyperparameter configuration (either from disk or using argparse with defaults in code, imo disk is cleaner)
- Ability to plot and store training results

https://github.com/Akettle44/LLMs-What-Can-Pruning-Tell-Us/blob/main/src/training/train.py

Acceptance Criteria:
Implement all things mentioned in goal. Show that training runs properly. Note that this acceptance criteria pertains to the code. It is expected that a separate issue will be needed to get good results from the model.

**Comment by Akettle44** [comment]
Completed
